molgenis-data-import/src/main/java/org/molgenis/data/importer/ImportWizardController.java

```diff
@@ -127,28 +127,38 @@ public void addGroupEntityClassPermissions(@RequestParam String groupId, WebRequ
 			String param = "radio-" + entityClassId;
 			String value = webRequest.getParameter(param);
 			if (value != null
-					&& (value.equalsIgnoreCase(org.molgenis.security.core.Permission.READ.toString())
-							|| value.equalsIgnoreCase(org.molgenis.security.core.Permission.COUNT.toString()) || value
-								.equalsIgnoreCase(org.molgenis.security.core.Permission.WRITE.toString()))
 					&& (SecurityUtils.currentUserHasRole(SecurityUtils.AUTHORITY_ENTITY_WRITE_PREFIX
 							+ entityClassId.toUpperCase()) || userAccountService.getCurrentUser().getSuperuser()))
 			{
-
-				authority
-						.setMolgenisGroup(dataService.findOne(MolgenisGroup.ENTITY_NAME, groupId, MolgenisGroup.class));
-				authority.setRole(SecurityUtils.AUTHORITY_ENTITY_PREFIX + value.toUpperCase() + "_"
-						+ entityClassId.toUpperCase());
-				if (authority.getId() == null)
+				if ((value.equalsIgnoreCase(org.molgenis.security.core.Permission.READ.toString())
+						|| value.equalsIgnoreCase(org.molgenis.security.core.Permission.COUNT.toString()) || value
+							.equalsIgnoreCase(org.molgenis.security.core.Permission.WRITE.toString())))
+				{
+					authority.setMolgenisGroup(dataService.findOne(MolgenisGroup.ENTITY_NAME, groupId,
+							MolgenisGroup.class));
+					authority.setRole(SecurityUtils.AUTHORITY_ENTITY_PREFIX + value.toUpperCase() + "_"
+							+ entityClassId.toUpperCase());
+					if (authority.getId() == null)
+					{
+						authority.setId(UUID.randomUUID().toString());
+						dataService.add(GroupAuthority.ENTITY_NAME, authority);
+					}
+					else dataService.update(GroupAuthority.ENTITY_NAME, authority);
+				}
+				else if (value.equals("none"))
```

Comment:
Add org.molgenis.security.core.Permission.NONE to Permission?
and use it as value.equals(org.molgenis.security.core.Permission.NONE)